Content/Items/Beach/Acessories.SaltCogs.cs

```diff
@@ -0,0 +1,89 @@
+﻿using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Food;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Beach
+{
+	class SaltCogs : SmartAccessory
+	{
+		public override string Texture => AssetDirectory.Assets + "Items/Beach/" + Name;
+
+		public SaltCogs() : base("Salt Cogs", "Your sentries fire twice as fast for two seconds after placing them\n100% increased sentry placement speed") { }
+
+		public override void Load()
+		{
+			StarlightProjectile.PostAIEvent += SpeedUp;
+			StarlightProjectile.PostDrawEvent += DrawCog;
+			StarlightItem.UseTimeMultiplierEvent += PlaceSpeed;
+			StarlightItem.UseAnimationMultiplierEvent += PlaceAnimation;
+		}
+
+		public override void SafeSetDefaults()
+		{
+			Item.rare = ItemRarityID.Blue;
+			Item.value = Item.sellPrice(silver: 50);
+		}
+
+		private void SpeedUp(Projectile projectile)
+		{
+			Player owner = Main.player[projectile.owner];
+
+			if (owner != null && Equipped(owner))
+			{
+				if (projectile.timeLeft == (Projectile.SentryLifeTime - 1))
+					projectile.extraUpdates += 1;
+
+				if (projectile.timeLeft == (Projectile.SentryLifeTime - 240))
+					projectile.extraUpdates -= 1;
+			}
+		}
+
+		private void DrawCog(Projectile projectile, Color lightColor)
+		{
+			Player owner = Main.player[projectile.owner];
+
+			if (owner != null && Equipped(owner))
+			{
+				if (projectile.timeLeft >= (Projectile.SentryLifeTime - 240))
```

Comment:
especially when the 240 gets reused like this makes it easier to balance adjust in the future and gives more meaning to this hanging magic number